spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -35,10 +40,12 @@
  * resolves the channel from the bean factory and, if not present, creates a new channel
  * and adds it to the factory after binding it to the binder. The binder is optionally
  * determined with a prefix preceding a colon.
+ *
  * @author Mark Fisher
  * @author Gary Russell
+ * @author Ilayaperumal Gopinathan
  */
-public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver {
+public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver implements SmartLifecycle {
```

Comment:
It looks like we may not make `BinderAwareChannelResolver` as Bindable since it would create unresolvable circular reference.

For instance, while `BinderAwareChannelResolver` is required as a bean for destination resolver the `ChannelBindingAutoConfiguration` requires all the `Bindable`s in it.